File: civitasJAVA/src/civitas/CasillaImpuesto.java
package civitas;

import java.util.ArrayList;


public class CasillaImpuesto extends Casilla {
    private float importe;
    private String nombre;
    
    //Constructor
    CasillaImpuesto(float cantidad, String nombre)
    {
        super(nombre);
        importe = cantidad;
    }
    
    //CasillaImpuesto recibe a jugador
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if (jugadorCorrecto(actual, todos) == true)
        {
            
            this.informe(actual, todos);
            todos.get(actual).pagaImpuesto(importe);
        }
    }
    
    //toString
    @Override
    public String toString()
    {
        return super.toString()+ (importe > 0 ? "\n\tImporte = " + importe : "");
    }
}

File: civitasRUBY/lib/casilla_impuesto.rb
#encoding: utf-8

require_relative "casilla"

module Civitas
  class CasillaImpuesto < Casilla
    def initialize(_importe, _nombre)
      super(_nombre)
      @importe = _importe
    end
    
    #RecibeJugador de impuesto
    def recibeJugador(actual,todos)
      if(jugadorCorrecto(actual,todos))
        informe(actual,todos)
        todos.at(actual).pagaImpuesto(@importe)
      end
    end
  end
end